src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs

```diff
@@ -3,50 +3,62 @@
 
 using System;
 using System.IO;
-using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Rewrite.Internal;
 using Microsoft.AspNetCore.Rewrite.Internal.ApacheModRewrite;
+using Microsoft.Extensions.FileProviders;
 
 namespace Microsoft.AspNetCore.Rewrite
 {
     /// <summary>
-    /// Apache mod_rewrite extensions on top of the <see cref="RewriteOptions"/>
+    /// Extensions for adding Apache mod_rewrite rules to <see cref="RewriteOptions"/>
     /// </summary>
     public static class ApacheModRewriteOptionsExtensions
     {
         /// <summary>
-        /// Imports rules from a mod_rewrite file and adds the rules to current rules. 
+        /// Add rules from a IIS config file containing Url Rewrite rules
         /// </summary>
-        /// <param name="options">The Rewrite options.</param>
-        /// <param name="hostingEnvironment">The Hosting Environment</param>
-        /// <param name="filePath">The path to the file containing mod_rewrite rules.</param>
-        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, IHostingEnvironment hostingEnvironment, string filePath)
+        /// <param name="options">The <see cref="RewriteOptions"/></param>
+        /// <param name="filePath">The path to the file containing UrlRewrite rules.</param>
+        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, string filePath)
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
-            if (hostingEnvironment == null)
+            var fileProvider = ((IRewriteOptionsInfrastructure)options).DefaultConfigFileProvider;
+            return options.AddApacheModRewrite(fileProvider, filePath);
+        }
+
+        /// <summary>
+        /// Add rules from an Apache mod_rewrite file
+        /// </summary>
+        /// <param name="options">The <see cref="RewriteOptions"/></param>
+        /// <param name="fileProvider">The <see cref="IFileProvider"/> </param>
+        /// <param name="filePath">The path to the file containing mod_rewrite rules.</param>
+        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, IFileProvider fileProvider, string filePath)
+        {
+            if (options == null)
             {
-                throw new ArgumentNullException(nameof(hostingEnvironment));
+                throw new ArgumentNullException(nameof(options));
             }
 
-            if (string.IsNullOrEmpty(filePath))
+            if (fileProvider == null)
             {
-                throw new ArgumentException(nameof(filePath));
+                throw new ArgumentNullException(nameof(fileProvider));
             }
 
-            var path = Path.Combine(hostingEnvironment.ContentRootPath, filePath);
-            using (var stream = File.OpenRead(path))
+            var fileInfo = fileProvider.GetFileInfo(filePath);
```

Comment:
Doesn't really matter. The next line will throw in that situation `fileInfo.CreateReadStream`